osgp/platform/osgp-secret-management/src/main/resources/db/migration/V20210823152146000__resize_column_secret_type.sql

```diff
@@ -0,0 +1,17 @@
+DO
+$$
+BEGIN
+
+IF  EXISTS (SELECT 1 FROM information_schema.columns
+  WHERE table_schema=current_schema AND table_name = 'encrypted_secret' AND column_name = 'secret_type') THEN
+  
+    DROP INDEX encrypted_secret_ix_secret_type;
+  
+ 	ALTER TABLE "encrypted_secret" ALTER COLUMN "secret_type" type VARCHAR(42);
+	
+	CREATE INDEX encrypted_secret_ix_secret_type ON encrypted_secret (secret_type);
```

Comment:
When making a varchar column larger I wouldn't expect issues, or a need to drop and rebuild indexes, am I wrong?

Indentation looks odd. Only using spaces for indentation and not mixing in tabs would fix this.

Going for a length of 42 would fix the issue for when a value for the new secret type `G_METER_FIRMWARE_UPDATE_AUTHENTICATION_KEY` would be inserted. When a new enum value is added that is a little larger, the issue would need to be fixed again.
Any reason not to jump to something like `varchar(255)`. I doubt we use this consistently, but it is used as the column length in for instance the osgp_core device_function_mapping table for the function_group and function columns that hold enum names as values.

Naming conventions for Flyway scripts recommend against ending with a series of zeroes. Though I have seen worse, avoid sticking some number of zeroes at the end as this is a habit people take over and appears hard to put an end to.

Reply:
- I really hope that the length of 255 is not applied for enumeration values. Values like `G_METER_FIRMWARE_UPDATE_AUTHENTICATION_KEY_WHICH_IS_NOT_THE_KEY_TO_BE_USED_FOR_OTHER_PURPOSES_TO_DO_AUTHENTICATION_WHILE_HAVING_THE_INTENTION_THAN_DOING_SOME_OTHER_THINGS_THAN_UPDATING_THE_FIRMWARE_ON_THE_DEVICE_FOR_WHICH_THIS_KEY_WAS_MENTION_FOR_CAPICE` will not contribute to readable code.

- The actions on the index are not neccessary. They are removed
- Flyway renamed leaving the zeroes out
- Indentation fixed